packages/nextjs/components/LinkGithub.tsx

```
@@ -28,8 +28,40 @@ const Username = ({ username }: { username: string }) => {
   );
 };
 
+const postGithubSignin = async ({
+  address,
+  username,
+  force = false,
+}: {
+  address: Address;
+  username: string;
+  force?: boolean;
+}) => {
+  const res = await fetch("/api/github", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({
+      address,
+      username,
+      force,
+    }),
+  });
+  const data = await res.json();
+  if (!res.ok) {
+    return {
+      status: res.status,
+      data: {},
```

Found a small issue here, the relink modal shows “already linked to address .” because the POST helper discards the error JSON, so `existingAddress` is undefined. We can return the parsed JSON even when !res.ok:

```
if (!res.ok) {
    return {
      status: res.status,
      data,
    };
  }
  ```
Let's make sure it doesn't have some bad side effects, in that case we can always just say "This GitHub account is already linked to another address." and not mention it.